File: devorbit-api/src/main/java/vn/edu/uit/devorbit_api/dto/student/GroupTaskResponse.java
package vn.edu.uit.devorbit_api.dto.student;

import vn.edu.uit.devorbit_api.entity.GroupTask;

import java.time.LocalDate;

public record GroupTaskResponse(
    Long id,
    Long groupPlanId,
    String title,
    String description,
    String assignedTo,
    LocalDate deadline,
    boolean completed,
    String createdBy,
    boolean deleteRequested,
    String deleteRequestedBy,
    String createdAt,
    String updatedAt
) {
    public static GroupTaskResponse from(GroupTask t) {
        return new GroupTaskResponse(
            t.getId(),
            t.getGroupPlan().getId(),
            t.getTitle(),
            t.getDescription(),
            t.getAssignedTo(),
            t.getDeadline(),
            t.isCompleted(),
            t.getCreatedBy(),
            t.isDeleteRequested(),
            t.getDeleteRequestedBy(),
            t.getCreatedAt() != null ? t.getCreatedAt().toString() : null,
            t.getUpdatedAt() != null ? t.getUpdatedAt().toString() : null
        );
    }
}

File: devorbit-mobile/app/src/main/java/vn/edu/uit/devorbit/mobile/data/remote/dto/GroupTaskResponse.kt
package vn.edu.uit.devorbit.mobile.data.remote.dto

data class GroupTaskResponse(
    val id: Long,
    val groupPlanId: Long,
    val planTitle: String?,
    val title: String,
    val description: String?,
    val assignedTo: String?,
    val deadline: String?,
    val completed: Boolean,
    val createdBy: String,
    val deleteRequested: Boolean,
    val deleteRequestedBy: String?,
    val createdAt: String?,
    val updatedAt: String?
)
